### PR TITLE
Translating ch12-03 in French.

### LINKED ISSUE #136: Translating ch12-03
I'm working on the French translation of this page.

### PR DESCRIPTION
**Pull request by Jimskapt**
Closes #136

### CODE REVIEW

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 1]
```diff
@@ -0,0 +1,1308 @@
...
+that remains in *main.rs* will be small enough to verify its correctness by
+reading it. Let’s rework our program by following this process.
+-->
+
+Cette structure permet de séparer les responsabilités : *main.rs* se charge de
+lancer le programme, et *lib.rs* renferme toute la logique des tâches à
+accomplir. Comme vous ne pouvez pas directement tester la fonction `main`, cette
+structure vous permet de tester toute la logique de votre programme en les
+déplaçant dans des fonctions dans *lib.rs*. Le seul code qui restera dans le
+*main.rs* sera suffisamment petit pour s'assurer qu'il soit correcte en le
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:167]:
```suggestion
*main.rs* sera suffisamment petit pour s'assurer qu'il soit correct en le
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 2]
```diff
@@ -0,0 +1,1308 @@
...
+
+<!--
+This rework may seem like overkill for our small program, but we’re refactoring
+in small, incremental steps. After making this change, run the program again to
+verify that the argument parsing still works. It’s good to check your progress
+often, to help identify the cause of problems when they occur.
+-->
+
+Ce remaniement peut sembler excessif pour notre petit programme, mais nous
+remanions avec de manière incrémentale par de petites étapes. Après avoir fait
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:245]:
```suggestion
remanions de manière incrémentale par de petites étapes. Après avoir fait
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 3]
```diff
@@ -0,0 +1,1308 @@
...
+We can take another small step to improve the `parse_config` function further.
+At the moment, we’re returning a tuple, but then we immediately break that
+tuple into individual parts again. This is a sign that perhaps we don’t have
+the right abstraction yet.
+-->
+
+Nous pouvons appliquer une nouvelle petite étape pour améliorer la fonction
+`interpreter_config`. Pour le moment, nous retournons un tuple, mais ensuite
+nous divisons immédiatement ce tuple à nouveau en plusieurs éléments. C'est un
+signe que nous n'avons peut-être la bonne approche.
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:267]:
```suggestion
signe que nous n'avons peut-être pas la bonne approche.
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 4]
```diff
@@ -0,0 +1,1308 @@
...
+> ### Les contre-parties de l'utilisation de `clone`
+>
+> Il y a une tendance chez les Rustacés de s'interdire l'utilisation de `clone`
+> pour régler les problèmes d'appartenance à cause du coût à l'exécution. Dans
+> le [chapitre 13][ch13]<!-- ignore -->, vous allez apprendre à utiliser des
+> méthodes plus efficaces dans ce genre de situation. Mais pour le moment, ce
+> n'est pas un problème de copier quelques chaînes de caractères pour continuer
+> à progresser car vous allez le faire une seule fois et votre `nom_fichier` et
+> `recherche` sont très courts. Il est plus important d'avoir un programme
+> fonctionnel qui est n'est très optimisé plutôt que d'essayer d'optimiser à
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:397]:
```suggestion
> fonctionnel qui n'est pas très optimisé plutôt que d'essayer d'optimiser à
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 5]
```diff
@@ -0,0 +1,1308 @@
...
+```rust,ignore
+{{#rustdoc_include ../listings-sources/ch12-an-io-project/listing-12-08/src/main.rs:here}}
+```
+-->
+
+```rust,ignore
+{{#rustdoc_include ../listings/ch12-an-io-project/listing-12-08/src/main.rs:here}}
+```
+
+```rust,ignore
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:583]: Ce bloc de code apparaît deux fois

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 6]
```diff
@@ -0,0 +1,1308 @@
...
+à `panic!`.
+
+<!--
+Listing 12-9 shows the changes we need to make to the return value of
+`Config::new` and the body of the function needed to return a `Result`. Note
+that this won’t compile until we update `main` as well, which we’ll do in the
+next listing.
+-->
+
+L'encart 12-9 nous montre les changements que nous devons faire pour à la
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:690]:
```suggestion
L'encart 12-9 nous montre les changements que nous devons apporter à la
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 7]
```diff
@@ -0,0 +1,1308 @@
...
+valeur de retour de `Config::new` et le corps de la fonction qui doit retourner
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:691]:
```suggestion
valeur de retour de `Config::new` et le corps de la fonction pour pouvoir retourner
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 8]
```diff
@@ -0,0 +1,1308 @@
...
+
+<!--
+Returning an `Err` value from `Config::new` allows the `main` function to
+handle the `Result` value returned from the `new` function and exit the process
+more cleanly in the error case.
+-->
+
+Retourner une valeur `Err` à partir de `Config::new` permet à la fonction
+`main` de gérer la valeur `Result` retournée par la fonction `new` et de
+terminer plus proprement le processus dans cas d'une erreur.
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:755]:
```suggestion
terminer plus proprement le processus dans le cas d'une erreur.
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 9]
```diff
@@ -0,0 +1,1308 @@
...
+We’ve added a new `use` line to bring `process` from the standard library into
+scope. The code in the closure that will be run in the error case is only two
+lines: we print the `err` value and then call `process::exit`. The
+`process::exit` function will stop the program immediately and return the
+number that was passed as the exit status code. This is similar to the
+`panic!`-based handling we used in Listing 12-8, but we no longer get all the
+extra output. Let’s try it:
+-->
+
+Nous avons ajouté une nouvelle ligne `use` pour importer `porté` dans la portée
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:848]:
```suggestion
Nous avons ajouté une nouvelle ligne `use` pour importer `process` dans la portée
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 10]
```diff
@@ -0,0 +1,1308 @@
...
+
+```rust,ignore
+// -- partie masquée ici --
+fn new(args: &[String]) -> Config {
+    if args.len() < 3 {
+        panic!("il n'y a pas assez d'arguments");
+    }
+    // -- partie masquée ici --
+```
+
```
Comment (Jimskapt) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:591]:
```suggestion
```

linked to @yozhgoor suggestion in https://github.com/Jimskapt/rust-book-fr/pull/137#r672536850

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 11]
```diff
@@ -0,0 +1,1308 @@
...
+for “dynamic.”
+-->
+
+En ce qui concerne le type d'erreur, nous avons utilisé *l'objet trait*
+`Box<dyn Error>` (et nous avons importé `std::error::Error` dans la portée avec
+une instruction `use` en haut). Nous allons voir les objets trait dans le
+[chapitre 17][ch17]<!-- ignore -->. Pour l'instant, retenez juste que
+`Box<dyn Error>` signifie que la fonction va retourner un type qui implémente
+le trait `Error`, mais que nous n'avons pas à préciser quel sera précisément le
+type de la valeur de retour. Cela nous donne de la flexibilité sur les valeurs
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:1012]:
```suggestion
type de la valeur de retour. Cela nous donne la flexibilité de retourner des valeurs
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 12]
```diff
@@ -0,0 +1,1308 @@
...
+d'erreurs de retour qui peuvent être différentes dans différents cas d'erreurs.
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:1013]:
```suggestion
qui peuvent être de différents types dans différents cas d'erreurs 
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 13]
```diff
@@ -0,0 +1,1308 @@
...
+may be of different types in different error cases. The `dyn` keyword is short
+for “dynamic.”
+-->
+
+En ce qui concerne le type d'erreur, nous avons utilisé *l'objet trait*
+`Box<dyn Error>` (et nous avons importé `std::error::Error` dans la portée avec
+une instruction `use` en haut). Nous allons voir les objets trait dans le
+[chapitre 17][ch17]<!-- ignore -->. Pour l'instant, retenez juste que
+`Box<dyn Error>` signifie que la fonction va retourner un type qui implémente
+le trait `Error`, mais que nous n'avons pas à préciser quel sera précisément le
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:1011]:
```suggestion
le trait `Error`, mais que nous n'avons pas à spécifier quel sera précisément le
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 14]
```diff
@@ -0,0 +1,1308 @@
...
+it doesn’t return a value we need.
+-->
+
+Troisièmement, la fonction `run` retourne maintenant une valeur `Ok` dans les
+cas de succès. Nous avons déclaré dans la signature que le type de succès de la
+fonction `run` était `()`, ce qui signifie que nous avons envelopper la valeur
+de type unité dans la valeur `Ok`. Cette syntaxe `Ok(())` peut sembler un peu
+étrange au départ, mais utiliser `()` de cette manière est la façon idéale
+d'indiquer que nous appelons `run` uniquement pour ses effets secondaires ; elle
+ne retourne pas de valeur que nous pourrions avoir besoin.
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:1043]:
```suggestion
ne retourne pas de valeur dont nous pourrions avoir besoin.
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 15]
```diff
@@ -0,0 +1,1308 @@
...
+We use `if let` rather than `unwrap_or_else` to check whether `run` returns an
+`Err` value and call `process::exit(1)` if it does. The `run` function doesn’t
+return a value that we want to `unwrap` in the same way that `Config::new`
+returns the `Config` instance. Because `run` returns `()` in the success case,
+we only care about detecting an error, so we don’t need `unwrap_or_else` to
+return the unwrapped value because it would only be `()`.
+-->
+
+Nous utilisons `if let` plutôt que `unwrap_or_else` pour vérifier si `run`
+retourne un valeure `Err` et appeler `process::exit(1)` le cas échéant. La
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:1116]:
```suggestion
retourne un valeur `Err` et appeler `process::exit(1)` le cas échéant. La
```

File: FRENCH/src/ch12-03-improving-error-handling-and-modularity.md [hunk 16]
```diff
@@ -0,0 +1,1308 @@
...
+
+<!--
+Our `minigrep` project is looking good so far! Now we’ll split the
+*src/main.rs* file and put some code into the *src/lib.rs* file so we can test
+it and have a *src/main.rs* file with fewer responsibilities.
+-->
+
+Notre projet `minigrep` se présente plutôt bien pour le moment ! Maintenant,
+nous allons diviser notre fichier *src/main.rs* et déplacer du code dans le
+fichier *src/lib.rs* pour que nous puissions le tester et avoir fichier
```
Comment (yozhgoor) [inline FRENCH/src/ch12-03-improving-error-handling-and-modularity.md:1146]:
```suggestion
fichier *src/lib.rs* pour que nous puissions le tester et avoir un fichier
```

**Pull request merged**